servers/rendering/renderer_rd/storage_rd/render_scene_buffers_rd.cpp

```diff
@@ -171,16 +171,18 @@ void RenderSceneBuffersRD::configure(const RenderSceneBuffersConfiguration *p_co
 	const bool resolve_target = msaa_3d != RS::VIEWPORT_MSAA_DISABLED;
 	create_texture(RB_SCOPE_BUFFERS, RB_TEX_COLOR, base_data_format, get_color_usage_bits(resolve_target, false, can_be_storage));
 
+	const uint32_t extra_bits = RD::TEXTURE_USAGE_TRANSIENT_BIT;
+
 	// Create our depth buffer.
-	create_texture(RB_SCOPE_BUFFERS, RB_TEX_DEPTH, get_depth_format(resolve_target, false, can_be_storage), get_depth_usage_bits(resolve_target, false, can_be_storage));
+	create_texture(RB_SCOPE_BUFFERS, RB_TEX_DEPTH, get_depth_format(resolve_target, false, can_be_storage), get_depth_usage_bits(resolve_target, false, can_be_storage) | extra_bits);
```

Comment:
I don't think this can always be transient as we can end up reading from it in two situations:
1. The user reads from the depth texture in a GDShader
2. Various effects use it (SDFGI, VoxelGI, SSAO, SSIL, SSR, DoF etc.)

Reply:
Makes sense.

We should probably:

1. Add basic validation from our side to prevent invalid operations (even on non-tilers, so that users can catch the bug on their machines).
2. In another PR, evaluate whether the flag can be enabled. Leave this PR to add the functionality without setting the flag.

Reply:
That makes sense!

Reply:
Something about this was bothering me a lot, and finally put a huge improvement to it.

The problem is that one could ask for TRANSIENT flag erroneously, and not notice it is wrong until you deploy to Vulkan or Metal (and have Validation scream at you, or see that everything is broken).

This wrong usage needs to be detected on Desktop PCs, regardless of whether the GPU is TBDR.

I've added the following:

```cpp
#ifdef DEBUG_ENABLED
					if (((const TextureInfo *)uniform.ids[j * 2 + 1].id)->transient) {
						ERR_PRINT("TEXTURE_USAGE_TRANSIENT_BIT texture must not be used for sampling in a shader.");
					}
#endif
```

and:

```cpp
#ifdef DEBUG_ENABLED
	if (src_tex_info->transient) {
		ERR_PRINT("TEXTURE_USAGE_TRANSIENT_BIT p_src_texture must not be used in command_copy_texture.");
	}
	if (dst_tex_info->transient) {
		ERR_PRINT("TEXTURE_USAGE_TRANSIENT_BIT p_dst_texture must not be used in command_copy_texture.");
	}
#endif
```

Where needed.

The rationale is as follows:

1. It's behind DEBUG_ENABLED because in some cases (not all of them) it may be expensive enough to do this check in the exported game. But it should be done while in Editor builds (when it is paramount to detect it)
2. It must not fail / early out. The most likely scenario is that the user is running the Editor on a Desktop GPU (unless you're on an ARM mac). The GPU command will render fine. But the editor will show you the error. This is IMO the best user-friendly outcome: rendering happens the way user intended; but the user is notified something is wrong.
3. User will notice this error early in his big monster machine. Not late when he decides to deploy to a phone (and not all phones, e.g. RDNA phones are not tilers so the user won't notice anything wrong).

I've tested this and Mobile does not trigger the error, but Forward+ of course does; because it is as you say: SSAO and others need to sample from the Depth Buffer.

Reply:
Nice! I appreciate the extra error validation. 

We should still remove the flag before merging though